src/shogun2-core/src/main/java/de/terrestris/shogun2/security/access/Shogun2PermissionEvaluator.java

```diff
@@ -42,7 +57,12 @@ public boolean hasPermission(Authentication authentication,
 				&& targetDomainObject instanceof PersistentObject
 				&& permissionObject instanceof String) {
 
-			final User user = (User) authentication.getPrincipal();
+			// get the user state when the user logged in
+			User user = (User) authentication.getPrincipal();
```

Comment:
I don't think so, but this should not be problem here as we check for `authentication.getPrincipal() instanceof User` a few lines before